.github/workflows/review.yaml

```diff
@@ -0,0 +1,22 @@
+name: 'Automagic Review'
+on:
+  pull_request:
+    types:
+      - opened
+    branches:
+      - main
+
+permissions:
+  contents: read
+  pull-requests: write
+
+jobs:
+  generate-review:
+    runs-on: ubuntu-latest
```

Comment:
### maintainability: It's a good practice to lock specific versions of actions. I see you are using `actions/checkout@v4`.  Have you considered pinning it to a specific version (like `actions/checkout@v2.4.0`) instead to prevent breaking changes from affecting your workflow in the future?